src/wasm/activity-service/activity/fit/session.go

```diff
@@ -155,3 +158,131 @@ func NewSession(mesg proto.Message) *activity.Session {
 
 	return ses
 }
+
+func convertSessionToMesg(ses *activity.Session) proto.Message {
+	mesg := factory.CreateMesgOnly(mesgnum.Session)
+
+	if !ses.Timestamp.IsZero() {
+		field := factory.CreateField(mesgnum.Session, fieldnum.SessionTimestamp)
+		field.Value = datetime.ToUint32(ses.Timestamp)
+		mesg.Fields = append(mesg.Fields, field)
+	}
+	if !ses.StartTime.IsZero() {
+		field := factory.CreateField(mesgnum.Session, fieldnum.SessionStartTime)
+		field.Value = datetime.ToUint32(ses.StartTime)
+		mesg.Fields = append(mesg.Fields, field)
+	}
+	if ses.Sport != "" {
+		sport := typedef.SportFromString(kit.FormatToLowerSnakeCase(ses.Sport))
+		if sport == typedef.SportInvalid {
+			sport = typedef.SportGeneric
+		}
+		field := factory.CreateField(mesgnum.Session, fieldnum.SessionSport)
+		field.Value = uint8(sport)
+		mesg.Fields = append(mesg.Fields, field)
+	}
+	if ses.TotalMovingTime != 0 {
+		field := factory.CreateField(mesgnum.Session, fieldnum.SessionTotalMovingTime)
+		field.Value = scaleoffset.DiscardAny(ses.TotalMovingTime, field.Type.BaseType(), field.Scale, field.Offset)
+		mesg.Fields = append(mesg.Fields, field)
+	}
+	if ses.TotalElapsedTime != 0 {
+		field := factory.CreateField(mesgnum.Session, fieldnum.SessionTotalElapsedTime)
+		field.Value = scaleoffset.DiscardAny(ses.TotalElapsedTime, field.Type.BaseType(), field.Scale, field.Offset)
+		mesg.Fields = append(mesg.Fields, field)
+
+		totalTimerTimeField := factory.CreateField(mesgnum.Lap, fieldnum.SessionTotalElapsedTime)
+		totalTimerTimeField.Value = field.Value
+		mesg.Fields = append(mesg.Fields, totalTimerTimeField)
+	}
+	if ses.TotalDistance != 0 {
+		field := factory.CreateField(mesgnum.Session, fieldnum.SessionTotalDistance)
+		field.Value = scaleoffset.DiscardAny(ses.AvgAltitude, field.Type.BaseType(), field.Scale, field.Offset)
```

Comment:
how is TotalDistance relate with AvgAltitude?

Reply:
ah you found a bug, it should be ses.TotalDistance